target/linux/ath79/generic/base-files/lib/upgrade/platform.sh

```diff
@@ -67,6 +67,13 @@ platform_do_upgrade() {
 		ROOTFS_FILE="root.squashfs"
 		platform_do_upgrade_failsafe_datachk "$1"
 		;;
+	huawei,ap5030dn)
+		# Write beginning address of "firmware" partition as KernelA address and KernelB address, each to BootupA & BootupB
+		echo "Setting Huawei-custom kernel addresses..."
+		echo -n -e "\x9e\x10\x00\x00\x9e\x10\x00\x00" | dd of=/dev/mtdblock1 bs=1 seek=$((0x254)) conv=notrunc
+		echo -n -e "\x9e\x10\x00\x00\x9e\x10\x00\x00" | dd of=/dev/mtdblock2 bs=1 seek=$((0x254)) conv=notrunc
+		default_do_upgrade "$1"
+		;;
```

Comment:
Is this the offset the Huawei bootloader uses to determine the firmware offset to boot from? You can use mtd lookup by name here nevertheless.

Reply:
Also drop the echo. I assume this is a debug-leftover

Reply:
Yes it is. Added it to the comment